test/unit/audited_hacks_test.rb

```diff
@@ -25,4 +25,114 @@ def test_user
     assert_equal User.current, audit.user
   end
 
+  class LoggingTest < ActiveSupport::TestCase
+    disable_transactional_fixtures!
+
+    setup do
+      @provider = FactoryBot.create(:provider_account)
```

Comment:
Imo it is better to change these 2 lines for:
```ruby
@provider = FactoryBot.create(:simple_provider)
User.current = FactoryBot.create(:admin, account: provider)
```

I have tested it locally and it works.
This way it is more efficient, because `provider_account` creates many more things that this test doesn't need 😄

Reply:
I have a different opinion about that. I think `simple_provider` shouldn't even exist. IMO, it only makes sense for unit tests of the account itself. Otherwise, it may bend the tests into a direction of, at some point, not even reflecting reality anymore. Unfortunately our code is too coupled, there are multiple things that go on deep down underneath, too many dependencies a given entity to exist in production, and there's no good separation of types of tests. Creating the bare minimum for a test to pass is (for now still) risky.

Reply:
Then we should have something like `:simple_provider_with_admin`. Because most of the time those 2 are coupled but that's it, we don't need to create buyers, services, plans, applications, etc for every single test and imo making tests even slower is very dangerous and i would avoid creating more objects than necessary.